site/content/dev-guide/custom_http_adapter.md

```diff
@@ -0,0 +1,109 @@
++++
+title = "Implement a Custom Hono HTTP Protocol Adapter"
+weight = 100
++++
+
+Hono comes with the Hono REST protocol adapter which you may use as a blueprint for your own HTTP protocol adapter 
+implementation. 
+<!--more-->
+
+
+
+This section will guide you through the steps to build your own custom Hono HTTP protocol adapter.
+
+## Prerequisites
+
+You should be familiar with the setup and start of Hono. Refer to the 
+[Getting Started]({{< relref "getting-started.md" >}}) guide.
+
+## The Hono REST Protocol Adapter
+
+The Hono REST protocol adapter supports telemetry and event data processing. Additionally, it provides an interface for 
+device registration. 
+
+For documentation of the Hono REST protocol adapter, refer to 
+[REST Adapter]({{< relref "component/rest-adapter.md" >}}) 
+
+You can find the source of the Hono REST protocol adapter at 
+<https://github.com/eclipse/hono/tree/master/adapters/rest-vertx>.
+
+## Anatomy of the Hono REST Protocol Adapter
+ 
+Like many other Hono components, the Hono REST protocol adapter makes use of the [Vert.x](https://vertx.io) tool-kit. 
+All Hono protocol adapters are implemented as Vert.x verticles.
+
+The Hono REST protocol adapter class `VertxBasedRestProtocolAdapter` is derived from an abstract base class. This base 
+class implements the functionality for component initialization, receiving telemetry and event data from devices or 
+external clients, and forwarding of data to the Hono server. Additionally, the Hono REST protocol adapter implements an 
+interface for device registration.
+
+## Derive Your Custom HTTP Adapter
+
+Use the Hono REST protocol adapter as a blueprint. If not desired, you may remove the methods implementing the 
+functionality for device registration.
+
+## Extend Your Custom HTTP Adapter
+
+### Adding Routes
+In Vert.x, a *route* is a mapping of a HTTP request to a handler. Inside a route, Vert.x provides a `RoutingContext` 
+instance which gives access to the HTTP request (and response) object containing the HTTP headers. 
+
+The Hono REST protocol adapter overrides the abstract method `addRoutes()`, provided by its base class, and adds the 
+routes for telemetry data, event data, and device registration.
+
+```java
+// route for uploading telemetry data
+router.route(HttpMethod.PUT, String.format("/telemetry/:%s/:%s", PARAM_TENANT, PARAM_DEVICE_ID))
+    .handler(ctx -> uploadTelemetryMessage(ctx, getTenantParam(ctx), getDeviceIdParam(ctx)));
+```
+ 
+The route for telemetry data parses the HTTP request, extracts the two parameters *tenant* and *deviceId* from the 
+request URL path, and forwards the message payload to the method `uploadTelemetryMessage()`, provided by the base class.
+
+{{% note %}}
+Note the Vert.x placeholder indicators `:` inside the URL path pattern `/telemetry/:%s/:%s`. 
```

Comment:
Why do I need to note them? I think you should explain what makes them special or important to note.